pkg/cpu/cpu_darwin.go

```diff
@@ -0,0 +1,110 @@
+package cpu
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var sysctlOutput = make(map[string]string)
+
+func (i *Info) load() error {
+	err := populateSysctlOutput()
+	if err != nil {
+		return errors.Wrap(err, "unable to populate sysctl map")
+	}
+
+	i.TotalCores = getTotalCores()
+	i.TotalThreads = getTotalThreads()
+	i.Processors = getProcessors()
+
+	return nil
+}
+
+// getProcessors some more info https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_system_capabilities
+func getProcessors() []*Processor {
+	p := make([]*Processor, getProcTopCount())
+	for i, _ := range p {
+		p[i] = new(Processor)
+		p[i].Vendor = sysctlOutput[fmt.Sprintf("hw.perflevel%s.name", strconv.Itoa(i))]
+		p[i].Model = getVendor()
+		p[i].NumCores = getNumberCoresFromPerfLevel(i)
+		p[i].Capabilities = getCapabilities()
+	}
+	return p
+}
+
+// valid for ARM see https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics
+func getCapabilities() []string {
+	caps := []string{}
+	for k, v := range sysctlOutput {
+		if strings.HasPrefix(k, "hw.optional.arm.") {
+			isEnabled, _ := strconv.Atoi(v)
+			if isEnabled == 1 {
+				caps = append(caps, k)
```

Comment:
This will make the CPU capability look like `hw.optional.arm.FEAT_BF16` instead of `FEAT_BF16`. Can we trim the `hw.optional.arm` part of the key?

Also, there are CPU capabilities that don't begin with `hw.optional.arm.`, including these important ones:

* `hw.optional.AdvSIMD_HPFPCvt`
* `hw.optional.AdvSIMD`
* `hw.optional.floatingpoint`
* `hw.optional.armv8_crc32`

Did we want to handle those?

Reply:
Yep totally makes sense! Clean the prefix and handling the extra ones